app/src/main/java/com/karumi/androidanimations/coordinatorlayout/ExerciseBehavior.kt

```diff
@@ -91,35 +95,36 @@ class ExerciseBehavior(
         /**
          * We are responsible for listening to new over-scrolls, that is, when dyUnconsumed is
          * positive: That means we got to the end of the list and still there is scrolling left
-         * to process. In that case we "simulate" the scroll by moving the whole view. When that
-         * happens onNestedPreScroll will start doing things because [target.top < 0].
+         * to process. In that case we "simulate" the scroll by translating the whole view. When that
+         * happens onNestedPreScroll will start doing things because [target.translateY < 0].
          */
-        if (dyUnconsumed > 0) {
-            overscroll(child, target, dyUnconsumed)
+        val isOverScrolling = dyUnconsumed > 0
+        if (isOverScrolling) {
+            handleScroll(child, target, dyUnconsumed)
         }
+        updateChildViewAnimationState(child)
     }
 
-    private fun overscroll(child: View, target: View, dy: Int) {
-        val targetHeight = target.measuredHeight
+    private fun handleScroll(child: View, target: View, dy: Int) {
+        updateTranslationY(target, dy)
+        updateTranslationY(child, dy)
+    }
 
-        /**
-         * We are going to calculate min and max positions for the bottom and the top points for
-         * both views. We are doing this so that small floating point operations do not carry
-         * errors around.
-         */
-        target.top = (target.top - dy).clamp(-childHeight, 0)
-        target.bottom = (target.bottom - dy).clamp(targetHeight - childHeight, targetHeight)
-        child.top = (child.top - dy).clamp(targetHeight - childHeight, targetHeight)
-        child.bottom = (child.bottom - dy).clamp(targetHeight, targetHeight + childHeight)
+    private fun updateTranslationY(view: View, dy: Int) {
+        val proposedTranslation = view.translationY - dy
+        view.translationY = proposedTranslation.clamp(-childHeight.toFloat(), 0f)
     }
 
-    private fun Int.clamp(min: Int, max: Int): Int {
-        return Math.min(max, Math.max(this, min))
+    private fun Float.clamp(min: Float, max: Float): Float {
```

Comment:
just a comment, as we are in kotlin why not: 

` private fun Float.clamp(min: Float, max: Float): Float = min(max, max(this, min))`

Reply:
Done!